java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture4/tasks/_2/Number2.java

```diff
@@ -0,0 +1,79 @@
+package com.zoolatech.lecture4.tasks._2;
+
+/*
+Create a class that represents a binary search tree.
+The tree should have a reference to the root node.
+Each node stores an integer value and has references to the left and right children nodes.
+The tree should also have a method that accepts a value and adds it to the tree.
+ */
+
+public class Number2 {
+    public static void main(String[] args) {
+        BinaryTree binaryTree = new BinaryTree();
+        BinaryTree.Node node = binaryTree.new Node(12);
+        binaryTree.insert(node, 15);
+        binaryTree.insert(node, 10);
+        binaryTree.insert(node, 20);
+        binaryTree.insert(node, 1);
+        binaryTree.insert(node, 13);
+        binaryTree.delete(node, 13);
+        binaryTree.delete(node, 10);
+        binaryTree.order(node);
+    }
+}
+
+class BinaryTree {
+
+    class Node {
```

Comment:
Does the class Node needs an access to some internal state of the BinaryTree class? Should it be defined as inner class?